src/main/java/net/openhft/hashing/LongTupleHashFunction.java

```diff
@@ -0,0 +1,779 @@
+ * other have default implementations which in the end delegate to
+ * {@link #hash(Object, Access, long, long, long[])} abstract method.
+ *
+ * The {@code #bitsLength()} method should also be implemented, returning the actual bits in the result
+ * array. The bits length should be greater than 64, otherwise just using the {@link #LongHashFunction}
+ * interface. And the length should also be a positive multiple of 8.
+ *
+ * <p>Notes about how exactly methods with default implementations are implemented in doc comments
+ * are given for information and could be changed at any moment. However, it could hardly cause
+ * any issues with subclassing, except probably little performance degradation. Methods documented
+ * as "shortcuts" could either delegate to the referenced method or delegate directly to the method
+ * to which the referenced method delegates.
+ *
+ *<p>{@code LongTupleHashFunction} implementations shouldn't assume that {@code Access} strategies
+ * do defensive checks, and access only bytes within the requested range.
+ */
+@ParametersAreNonnullByDefault
+public abstract class LongTupleHashFunction implements Serializable {
+    private static final long serialVersionUID = 0L;
+
+    // Implementations
+    //
+
+    /**
+     * Returns a 128-bit hash function implementing
+     * <a href="https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp">MurmurHash3
+     * algorithm</a> without seed values. This implementation produces equal results for equal input
+     * on platforms with different {@link ByteOrder}, but is slower on big-endian platforms than on
+     * little-endian.
+     *
+     * @see #murmur_3(long)
+     */
+    @NotNull
+    public static LongTupleHashFunction murmur_3() {
+        return MurmurHash_3.asLongTupleHashFunctionWithoutSeed();
+    }
+
+    /**
+     * Returns a 128-bit hash function implementing
+     * <a href="https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp">MurmurHash3
+     * algorithm</a> with the given seed value. This implementation produces equal results for equal
+     * input on platforms with different {@link ByteOrder}, but is slower on big-endian platforms
+     * than on little-endian.
+     *
+     * @see #murmur_3()
+     */
+    @NotNull
+    public static LongTupleHashFunction murmur_3(final long seed) {
+        return MurmurHash_3.asLongTupleHashFunctionWithSeed(seed);
+    }
+
+    /**
+     * Constructor for use in subclasses.
+     */
+    protected LongTupleHashFunction() {}
+
+    // Public API
+    //
+
+    /**
+     * Returns the number of bits in a result array; a positive multiple of 8.
+     */
+    public abstract int bitsLength();
+
+    /**
+     * Returns a result array
+     */
+    @NotNull
+    public long[] newResultArray() {
+        return new long[(bitsLength() + 63) / 64];
+    }
+
+    /**
+     * Returns the hash code for the given {@code long} value; this method is consistent with
```

Comment:
This method doesn't *return* anything, as well as all similar methods. Please adjust the language.